subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -244,9 +244,12 @@ static size_t gatt_prov_adv_create(struct bt_data prov_sd[2])
 
 dev_name:
 	if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT_USE_DEVICE_NAME)) {
+		uint8_t name[BT_GAP_DEVICE_NAME_MAX_SIZE];
```

Comment:
It will be nice to have `BT_GAP_DEVICE_NAME_MAX_SIZE` configured over KConfig despite it has specified size. I doubt in that all customers will consume all 237 bytes. Due to this, mesh functions will have stack memory overconsumption.

Reply:
I created `DEVICE_NAME_SIZE` in `mesh/gatt.h` which conditionally take the value of `BT_GAP_DEVICE_NAME_MAX_SIZE` or `CONFIG_BT_GAP_DEVICE_NAME_DYNAMIC_MAX` if `CONFIG_BT_GAP_DEVICE_NAME_DYNAMIC` is enabled.